scipy/stats/contingency.py

```diff
@@ -171,6 +171,7 @@ def chi2_contingency(observed, correction=True, lambda_=None):
     fisher_exact
     chisquare
     power_divergence
+    barnard_exact
```

Comment:
Oops. When I suggested this go in `hypotests`, I didn't know about `contingency.py`. That might make more sense, but don't change yet. @WarrenWeckesser thoughts on where this should go?

Reply:
Since there isn't any answer, what should I do?

Reply:
: ) 
Some reviews take a long time waiting for feedback. I think the thing to do is improve what we can and be patient for the rest!

Reply:
Actually please send an email to the developer mailing list if you haven't already. You need to propose new features there before we can add to the public API. Make a case for why the test is a good thing to add, and ask where it should go.

Reply:
Sorry to ask but: 

> eatures there before we can add to the public API. Make a case for why the test is a good thing to add (including resolving the request in gh-11094), and ask where it should go.
Do you mean #11014 ?

Reply:
Yes, oops!

Reply:
> Actually please send an email to the developer mailing list if you haven't already. You should propose new features there before we can add to the public API. Make a case for why the test is a good thing to add (including resolving the request in #11014, and ask where it should go.

@jb-leger will you help with this ?

Reply:
Fine, so, I have nothing to do :-)

Reply:
Hmm... I didn't receive it. What was the subject?
In any case, I talked to @WarrenWeckesser today, and he's going to send an email to the list asking whether we should add `fisher_exact` to the `scipy.stats.contingency` namespace, and he will ask about this PR in the email, too. This function should probably go wherever Fisher's test goes, so I think his email will also help us get the answer we need.